includes/admin/class-admin.php

```diff
@@ -515,9 +515,16 @@ public function admin_menu_combine() {
 			$cpt_menu   = 'edit.php?post_type=' . WPBDP_POST_TYPE;
 			$admin_menu = $this->menu_id;
 
-			if ( isset( $submenu[ $cpt_menu ] ) && isset( $submenu[ $admin_menu ] ) ) {
+			$current_user = wp_get_current_user();
+			$roles        = $current_user->roles;
+			$role         = ! empty( $roles ) && is_array( $roles ) ? array_shift( $roles ) : '';
+
+			if ( isset( $submenu[ $cpt_menu ] ) && ( isset( $submenu[ $admin_menu ] ) || $role === 'editor' ) ) {
```

Comment:
This looks like it could potentially have unwanted side effects. Can we move this to elseif instead?

This issue has been repeating. Can you also add unit tests to check for the expected menu items for different user roles?